### PR TITLE
keep the order of added list in add_columns

### LINKED ISSUE #13779: table.add_columns results in different order if no indexes are provided (with proposed solution)
### Description

If I add a list of columns to a table with `add_columns`, without specifying indexes, the added columns in resulting table will have different orders from the given list.

Although the order of the columns does not change much, but I would expect what I put in the table stays the same order. This will affect user reading the table data, since they will see the seemingly random order. 

**The exact behavior** can be found from the code:
https://github.com/astropy/astropy/blob/0b5c322d18c5cc910aa796720d9e1bb49a0a5404/astropy/table/table.py#L2307-L2323

If `indexes` is None, it will be assigned with `[len(self.columns)] * len(cols)`, which means always insert at the end of the existing table columns. Then, the provided columns are added by the order of `reversed(np.argsort(indexes))`.

The problem is in `np.argsort`, which uses quick sort by default. For a list of same values, the result will not be the same order of the given list. That's not visible from the resulting list, but visible from `argsort` result, which gives the indexes of the sorted list. Thus the issue.

**Proposed solution:**

If it does not affect other things, maybe use merge sort as the sorting method for `np.argsort`. Since merge sort don't (maybe?) change the original order if the items are all the same.

### Expected behavior

Those list will keep the same order.

### Actual behavior

The columns added in the table have different order.

### Steps to Reproduce

1. See the code

```python
import numpy as np
from astropy.table import Table
t = Table()
names = [str(i) for i in range(20)]
t.add_columns(np.random.rand(20, 2), names=names)
t.colnames
```

2. Shows the following list

```
['0', '17', '16', '15', '14', '13', '12', '11', '10', '9', '8', '7', '6', '5', '4', '3', '2', '1', '18', '19']
```

### System Details

Linux-5.19.6-arch1-1-x86_64-with-glibc2.36
Python 3.10.7 (main, Sep  6 2022, 21:22:27) [GCC 12.2.0]
Numpy 1.23.3
pyerfa 2.0.0.1
astropy 5.1
Scipy 1.9.1
Matplotlib 3.5.2

### PR DESCRIPTION
**Pull request by oliverlew**
### Description

When using `add_columns` without specifying indexes, the order will be given by numpy's argsort function. But by default, argsort uses quick sort, which changes the index order. This fix changes the sort method to "stable" to keep the original list order.

fixes #13779.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Thank you! 🙏

Can you please add a change log too? See https://github.com/astropy/astropy/blob/main/docs/changes/README.rst

I'll let @taldcroft do a thorough review and move the milestone if backport is not applicable for this case.

File: astropy/table/tests/test_table.py
Comment (taldcroft) [inline astropy/table/tests/test_table.py:660]: For this test there is no additional coverage provided by the three different `table_types` (they all wind up at the same base method). There are likely other tests in this class where that is also true, but for now you should just move this out of the class as a single test and do `t = Table()`.

**Review by taldcroft — COMMENTED** [review]
Looks great, thanks @oliverlew! There is just one tiny change in how the test is written.

**Comment by taldcroft** [comment]
@oliverlew - if you are comfortable with rebasing I would suggest squashing down to one commit since this is basically just a one-line code change. I can help if you haven't done this.

File: astropy/table/tests/test_table.py
Comment (oliverlew) [reply astropy/table/tests/test_table.py:660]: Thanks for clarifying. I saw those three table types during the test, but had no idea what that is for or when a test should use it. I will update the PR.

The standalone test functions don't seem to have specific sort order, so I just put the function at the end. Please let me know if anything is wrong.

**Comment by oliverlew** [comment]
Done, squashed into one commit.

**Comment by taldcroft** [comment]
Awesome, once this gets through CI I will approve and merge!

**Comment by pllim** [comment]
dev job failure is unrelated and you can ignore the pesky "CircleCI Pipeline". Thanks!

**Review by taldcroft — APPROVED** [review]
Thanks @oliverlew !!

**Comment by taldcroft** [comment]
Now the `Check PR change log / Check if towncrier change log entry is correct` job got canceled for some reason...

**Comment by pllim** [comment]
The change log job been weird lately but I don't have time to investigate... Merging. Thanks!

**Comment by pllim** [comment]
Hmm. Do you want to attempt manual backport or should I move the milestone?

**Comment by taldcroft** [comment]
Since it is not a critical bug and there is an easy workaround, please move the milestone and we will not backport to 5.0.x. Thanks!

**Pull request merged**
